substrate/frame/child-bounties/src/benchmarking.rs

```diff
@@ -174,145 +168,225 @@ fn assert_last_event<T: Config>(generic_event: <T as Config>::RuntimeEvent) {
+		pallet_treasury::Pallet::<T>::on_initialize(BlockNumberFor::<T>::zero());
 		frame_system::Pallet::<T>::set_block_number(T::BountyUpdatePeriod::get() + 1u32.into());
 		let caller = whitelisted_caller();
-	}: _(RawOrigin::Signed(caller), bounty_setup.bounty_id,
-			bounty_setup.child_bounty_id)
 
-	award_child_bounty {
+		#[extrinsic_call]
+		_(RawOrigin::Signed(caller), bounty_setup.bounty_id, bounty_setup.child_bounty_id);
+
+		Ok(())
+	}
+
+	#[benchmark]
+	fn award_child_bounty() -> Result<(), BenchmarkError> {
 		setup_pot_account::<T>();
 		let bounty_setup = activate_child_bounty::<T>(0, T::MaximumReasonLength::get())?;
-		let beneficiary_account: T::AccountId = account("beneficiary", 0, SEED);
+		let beneficiary_account = account::<T::AccountId>("beneficiary", 0, SEED);
 		let beneficiary = T::Lookup::unlookup(beneficiary_account.clone());
-	}: _(RawOrigin::Signed(bounty_setup.child_curator), bounty_setup.bounty_id,
-			bounty_setup.child_bounty_id, beneficiary)
-	verify {
-		assert_last_event::<T>(Event::Awarded {
-			index: bounty_setup.bounty_id,
-			child_index: bounty_setup.child_bounty_id,
-			beneficiary: beneficiary_account
-		}.into())
+
+		#[extrinsic_call]
+		_(
+			RawOrigin::Signed(bounty_setup.child_curator),
+			bounty_setup.bounty_id,
+			bounty_setup.child_bounty_id,
+			beneficiary,
+		);
+
+		assert_last_event::<T>(
+			Event::Awarded {
+				index: bounty_setup.bounty_id,
+				child_index: bounty_setup.child_bounty_id,
+				beneficiary: beneficiary_account,
+			}
+			.into(),
+		);
+
+		Ok(())
 	}
 
-	claim_child_bounty {
+	#[benchmark]
+	fn claim_child_bounty() -> Result<(), BenchmarkError> {
 		setup_pot_account::<T>();
 		let bounty_setup = activate_child_bounty::<T>(0, T::MaximumReasonLength::get())?;
-		let beneficiary_account: T::AccountId = account("beneficiary", 0, SEED);
+		let beneficiary_account = account("beneficiary", 0, SEED);
 		let beneficiary = T::Lookup::unlookup(beneficiary_account);
 
-		ChildBounties::<T>::award_child_bounty(
+		Pallet::<T>::award_child_bounty(
 			RawOrigin::Signed(bounty_setup.child_curator.clone()).into(),
 			bounty_setup.bounty_id,
 			bounty_setup.child_bounty_id,
-			beneficiary
+			beneficiary,
 		)?;
 
-		let beneficiary_account: T::AccountId = account("beneficiary", 0, SEED);
-		let beneficiary = T::Lookup::unlookup(beneficiary_account.clone());
+		let beneficiary_account = account("beneficiary", 0, SEED);
 
 		frame_system::Pallet::<T>::set_block_number(T::BountyDepositPayoutDelay::get());
-		ensure!(T::Currency::free_balance(&beneficiary_account).is_zero(),
-			"Beneficiary already has balance.");
-
-	}: _(RawOrigin::Signed(bounty_setup.curator), bounty_setup.bounty_id,
-			bounty_setup.child_bounty_id)
-	verify {
-		ensure!(!T::Currency::free_balance(&beneficiary_account).is_zero(),
-			"Beneficiary didn't get paid.");
+		assert!(
+			T::Currency::free_balance(&beneficiary_account).is_zero(),
+			"Beneficiary already has balance."
+		);
```

Comment:
ensure is better than assert, because the benchmarking pipeline can handle errors

Reply:
For example, it will output logs or data, whereas a panic will not

Reply:
Didn't think about this before.

Since I encountered a lot of benchmarking code, `assert!` is used more frequently. I will revert this and create an issue for discussion.